lazy_theta_star_planner/include/lazy_theta_star_planner/lazy_theta_star2.h

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LAZY_THETA_STAR_B__LAZY_THETA_STAR_H_
+#define LAZY_THETA_STAR_B__LAZY_THETA_STAR_H_
+
+#include <cmath>
+#include <vector>
+#include <queue>
+#include <algorithm>
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+
+typedef int id;
+typedef double cost;
+typedef int map_pts;
+typedef double world_pts;
+
+#define INF_COST DBL_MAX
+#define LETHAL_COST 252
+
+template<typename coordsT>
+struct coords
+{
+  coordsT x, y;
+};
+
+struct pos
+{
+  id pos_id;
+  cost * f;
```

Comment:
Could this not be a reference rather than a pointer?

Reply:
Generally the rule of thumb is use pointers when you just can't use references

Reply:
It would make the comp operator() cleaner

Reply:
I use a pointer specially, so as to not cause an error

Reply:
Why would there be an error? you can have a reference member to a struct, you just need to make a contructor to take it in and assign it

Reply:
There's a deletion of move assignment operator required by std::priority_queue, because, according the error, pos& operator=(pos&&) is implicitly deleted, as the default one is ill - formed. This would then require adding manually a move assignment constructor, should I do it?

Reply:
A pointer is fine then.